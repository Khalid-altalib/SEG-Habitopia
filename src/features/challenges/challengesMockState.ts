import { ChallengesState } from "./challengesSlice";

// Mocks the challenge state for testing purposes
const challengesMockState: ChallengesState = {
    challenges: [
        {
            id: '1',
            name: 'Fitness',
            description: 'Be active',
            active: true

        },
        {
            id: '1',
            name: 'Study',
            description: 'Study hard',
            active: false
        }
    ],
    fetchChallenges: {
        loading: false,
        error: ""
    },
    joinChallenge: {
        loading: false,
        error: ""
    }
};

export default challengesMockState;