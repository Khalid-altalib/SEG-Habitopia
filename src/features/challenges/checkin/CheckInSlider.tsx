import { HStack } from "native-base";
import React from "react";
import CheckInBox from "./CheckInBox";

type Props = {};

const CheckInSlider = (props: Props) => {
  const data = [
    {
      name: "Study",
      timeLeft: "3",
    },
    {
      name: "Sleep",
      timeLeft: "3",
    },
    {
      name: "Fitness",
      timeLeft: "3",
    },
    {
      name: "study",
      timeLeft: "3",
    },
    {
      name: "sleep",
      timeLeft: "3",
    },
    {
      name: "fitness",
      timeLeft: "3",
    },
    {
      name: "study",
      timeLeft: "3",
    },
    {
      name: "sleep",
      timeLeft: "3",
    },
    {
      name: "fitness",
      timeLeft: "3",
    },
  ];
  return (
    <HStack mt={3}>
      {data.map((checkIn, index) => (
        <CheckInBox key={index} checkIn={checkIn} />
      ))}
    </HStack>
  );
};

export default CheckInSlider;
