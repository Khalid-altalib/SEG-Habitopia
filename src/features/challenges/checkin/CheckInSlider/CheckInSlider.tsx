import { useDispatch, useSelector } from "@app/hooks";
import StatusContainer from "@components/StatusContainer/StatusContainer";
import Text from "@components/Text";
import { fetchCheckInSnippet } from "@features/chat/chatSlice";
import { Center, HStack } from "native-base";
import { ControlledPropUpdatedSelectedItem } from "native-base/lib/typescript/components/composites/Typeahead/useTypeahead/types";
import React, { useEffect } from "react";
import CheckInBox from "../CheckInBox/CheckInBox";

type Props = {};

const CheckInSlider = (props: Props) => {
  const { checkInSnippet, fetchCheckInSnippet: requestStatus } = useSelector(
    (state) => state.chats
  );

  const { loading, error } = requestStatus;

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchCheckInSnippet());
  }, []);

  return (
    <StatusContainer
      loading={loading}
      error={error}
      data={checkInSnippet}
      noDataDisplay={
        <Center width="100%">
          <Text
            style={{ textAlign: "center", marginLeft: -10, marginRight: 12 }}
          >
            All Caught Up!
          </Text>
        </Center>
      }
    >
      <HStack marginLeft={25} marginTop={25} maxHeight={150} minWidth="100%">
        {checkInSnippet.map((checkInSnippetItem) => (
          <CheckInBox
            key={checkInSnippetItem.chatId}
            checkInSnippetItem={checkInSnippetItem}
          />
        ))}
      </HStack>
    </StatusContainer>
  );
};

export default CheckInSlider;
