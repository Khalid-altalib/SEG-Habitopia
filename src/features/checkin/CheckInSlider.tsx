import { HStack } from "native-base";
import React from "react";
import CheckInBox from "./CheckInBox";

type Props = {};

const CheckInSlider = (props: Props) => {
  const data = [
    {
      name: "Study",
      timeLeft: "3",
    },
    {
      name: "Sleep",
      timeLeft: "3",
    },
    {
      name: "Fitness",
      timeLeft: "3",
    },
    {
      name: "Study",
      timeLeft: "3",
    },
    {
      name: "Sleep",
      timeLeft: "3",
    },
    {
      name: "Fitness",
      timeLeft: "3",
    },
    {
      name: "Study",
      timeLeft: "3",
    },
    {
      name: "Sleep",
      timeLeft: "3",
    },
    {
      name: "Fitness",
      timeLeft: "3",
    },
  ];
  return (
    <HStack marginLeft={25} marginTop={25} height={150}>
      {data.map((checkIn, index) => (
        <CheckInBox key={index} checkIn={checkIn} />
      ))}
    </HStack>
  );
};

export default CheckInSlider;
