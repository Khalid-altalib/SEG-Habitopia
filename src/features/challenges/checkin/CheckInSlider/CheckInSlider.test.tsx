import { render } from "@testing-library/react-native";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import TestingWrapperNavigation from "@app/testingWrapperWithNavigation";
import CheckInSlider from "./CheckInSlider";
import challengesMockState from "@features/challenges/challengesMockState";

describe("CheckInSlider", () => {
  const mockState = {
    challenges: challengesMockState,
    checkInSnippetItem: {
      challenge: challengesMockState.challenges[0],
      endDate: "",
      chatId: "",
    },
    sendCheckIn: {
      loading: false,
      error: "",
    },
    fetchCheckInSnippet: {
      loading: false,
      error: "",
    },
  };
  const mockStore = configureStore([thunk])(mockState);

  it("renders the check-in slider correctly", async () => {
    const component = render(
      <TestingWrapperNavigation store={mockStore}>
        <CheckInSlider />
      </TestingWrapperNavigation>
    );

    expect(component).toMatchSnapshot();
  });

  it("finds the check-in slider", async () => {
    const component = render(
      <TestingWrapperNavigation store={mockStore}>
        <CheckInSlider />
      </TestingWrapperNavigation>
    );

    const box = component.getByTestId("checkInSlider");

    expect(box).toBeDefined();
  });
});
