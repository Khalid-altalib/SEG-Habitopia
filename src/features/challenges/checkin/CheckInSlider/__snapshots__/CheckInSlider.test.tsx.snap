// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckInSlider renders the check-in slider correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="TestScreen"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <View
              dataSet={{}}
              style={
                {
                  "flexDirection": "row",
                  "marginTop": 12,
                }
              }
              testID="checkInSlider"
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
                testID="checkInBox"
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "paddingBottom": 20,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 20,
                      "width": 150,
                    }
                  }
                >
                  <Image
                    accessibilityLabel="Alternate Text"
                    alt="Alternate Text"
                    dataSet={{}}
                    onError={[Function]}
                    source={
                      {
                        "uri": "https://picsum.photos/200",
                      }
                    }
                    style={
                      {
                        "borderRadius": 12,
                        "height": 128,
                        "maxWidth": "100%",
                        "position": "absolute",
                        "width": 128,
                      }
                    }
                  />
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#171717",
                        "elevation": 2,
                        "fontFamily": undefined,
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "letterSpacing": 0,
                        "lineHeight": 20,
                        "marginLeft": 8,
                        "marginTop": 4,
                        "position": "absolute",
                        "shadowColor": "#000000",
                        "shadowOffset": {
                          "height": 1,
                          "width": 0,
                        },
                        "shadowOpacity": 0.2,
                        "shadowRadius": 1.41,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    Study
                  </Text>
                  <View
                    dataSet={{}}
                    style={
                      {
                        "marginLeft": -12,
                        "marginTop": 80,
                      }
                    }
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#dc2626",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      3
                      h left!
                    </Text>
                  </View>
                </View>
              </View>
              <View
                dataSet={{}}
                style={{}}
              />
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
                testID="checkInBox"
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "paddingBottom": 20,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 20,
                      "width": 150,
                    }
                  }
                >
                  <Image
                    accessibilityLabel="Alternate Text"
                    alt="Alternate Text"
                    dataSet={{}}
                    onError={[Function]}
                    source={
                      {
                        "uri": "https://picsum.photos/200",
                      }
                    }
                    style={
                      {
                        "borderRadius": 12,
                        "height": 128,
                        "maxWidth": "100%",
                        "position": "absolute",
                        "width": 128,
                      }
                    }
                  />
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#171717",
                        "elevation": 2,
                        "fontFamily": undefined,
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "letterSpacing": 0,
                        "lineHeight": 20,
                        "marginLeft": 8,
                        "marginTop": 4,
                        "position": "absolute",
                        "shadowColor": "#000000",
                        "shadowOffset": {
                          "height": 1,
                          "width": 0,
                        },
                        "shadowOpacity": 0.2,
                        "shadowRadius": 1.41,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    Sleep
                  </Text>
                  <View
                    dataSet={{}}
                    style={
                      {
                        "marginLeft": -12,
                        "marginTop": 80,
                      }
                    }
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#dc2626",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      3
                      h left!
                    </Text>
                  </View>
                </View>
              </View>
              <View
                dataSet={{}}
                style={{}}
              />
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
                testID="checkInBox"
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "paddingBottom": 20,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 20,
                      "width": 150,
                    }
                  }
                >
                  <Image
                    accessibilityLabel="Alternate Text"
                    alt="Alternate Text"
                    dataSet={{}}
                    onError={[Function]}
                    source={
                      {
                        "uri": "https://picsum.photos/200",
                      }
                    }
                    style={
                      {
                        "borderRadius": 12,
                        "height": 128,
                        "maxWidth": "100%",
                        "position": "absolute",
                        "width": 128,
                      }
                    }
                  />
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#171717",
                        "elevation": 2,
                        "fontFamily": undefined,
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "letterSpacing": 0,
                        "lineHeight": 20,
                        "marginLeft": 8,
                        "marginTop": 4,
                        "position": "absolute",
                        "shadowColor": "#000000",
                        "shadowOffset": {
                          "height": 1,
                          "width": 0,
                        },
                        "shadowOpacity": 0.2,
                        "shadowRadius": 1.41,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    Fitness
                  </Text>
                  <View
                    dataSet={{}}
                    style={
                      {
                        "marginLeft": -12,
                        "marginTop": 80,
                      }
                    }
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#dc2626",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      3
                      h left!
                    </Text>
                  </View>
                </View>
              </View>
              <View
                dataSet={{}}
                style={{}}
              />
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
                testID="checkInBox"
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "paddingBottom": 20,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 20,
                      "width": 150,
                    }
                  }
                >
                  <Image
                    accessibilityLabel="Alternate Text"
                    alt="Alternate Text"
                    dataSet={{}}
                    onError={[Function]}
                    source={
                      {
                        "uri": "https://picsum.photos/200",
                      }
                    }
                    style={
                      {
                        "borderRadius": 12,
                        "height": 128,
                        "maxWidth": "100%",
                        "position": "absolute",
                        "width": 128,
                      }
                    }
                  />
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#171717",
                        "elevation": 2,
                        "fontFamily": undefined,
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "letterSpacing": 0,
                        "lineHeight": 20,
                        "marginLeft": 8,
                        "marginTop": 4,
                        "position": "absolute",
                        "shadowColor": "#000000",
                        "shadowOffset": {
                          "height": 1,
                          "width": 0,
                        },
                        "shadowOpacity": 0.2,
                        "shadowRadius": 1.41,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    study
                  </Text>
                  <View
                    dataSet={{}}
                    style={
                      {
                        "marginLeft": -12,
                        "marginTop": 80,
                      }
                    }
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#dc2626",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      3
                      h left!
                    </Text>
                  </View>
                </View>
              </View>
              <View
                dataSet={{}}
                style={{}}
              />
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
                testID="checkInBox"
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "paddingBottom": 20,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 20,
                      "width": 150,
                    }
                  }
                >
                  <Image
                    accessibilityLabel="Alternate Text"
                    alt="Alternate Text"
                    dataSet={{}}
                    onError={[Function]}
                    source={
                      {
                        "uri": "https://picsum.photos/200",
                      }
                    }
                    style={
                      {
                        "borderRadius": 12,
                        "height": 128,
                        "maxWidth": "100%",
                        "position": "absolute",
                        "width": 128,
                      }
                    }
                  />
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#171717",
                        "elevation": 2,
                        "fontFamily": undefined,
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "letterSpacing": 0,
                        "lineHeight": 20,
                        "marginLeft": 8,
                        "marginTop": 4,
                        "position": "absolute",
                        "shadowColor": "#000000",
                        "shadowOffset": {
                          "height": 1,
                          "width": 0,
                        },
                        "shadowOpacity": 0.2,
                        "shadowRadius": 1.41,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    sleep
                  </Text>
                  <View
                    dataSet={{}}
                    style={
                      {
                        "marginLeft": -12,
                        "marginTop": 80,
                      }
                    }
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#dc2626",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      3
                      h left!
                    </Text>
                  </View>
                </View>
              </View>
              <View
                dataSet={{}}
                style={{}}
              />
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
                testID="checkInBox"
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "paddingBottom": 20,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 20,
                      "width": 150,
                    }
                  }
                >
                  <Image
                    accessibilityLabel="Alternate Text"
                    alt="Alternate Text"
                    dataSet={{}}
                    onError={[Function]}
                    source={
                      {
                        "uri": "https://picsum.photos/200",
                      }
                    }
                    style={
                      {
                        "borderRadius": 12,
                        "height": 128,
                        "maxWidth": "100%",
                        "position": "absolute",
                        "width": 128,
                      }
                    }
                  />
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#171717",
                        "elevation": 2,
                        "fontFamily": undefined,
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "letterSpacing": 0,
                        "lineHeight": 20,
                        "marginLeft": 8,
                        "marginTop": 4,
                        "position": "absolute",
                        "shadowColor": "#000000",
                        "shadowOffset": {
                          "height": 1,
                          "width": 0,
                        },
                        "shadowOpacity": 0.2,
                        "shadowRadius": 1.41,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    fitness
                  </Text>
                  <View
                    dataSet={{}}
                    style={
                      {
                        "marginLeft": -12,
                        "marginTop": 80,
                      }
                    }
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#dc2626",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      3
                      h left!
                    </Text>
                  </View>
                </View>
              </View>
              <View
                dataSet={{}}
                style={{}}
              />
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
                testID="checkInBox"
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "paddingBottom": 20,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 20,
                      "width": 150,
                    }
                  }
                >
                  <Image
                    accessibilityLabel="Alternate Text"
                    alt="Alternate Text"
                    dataSet={{}}
                    onError={[Function]}
                    source={
                      {
                        "uri": "https://picsum.photos/200",
                      }
                    }
                    style={
                      {
                        "borderRadius": 12,
                        "height": 128,
                        "maxWidth": "100%",
                        "position": "absolute",
                        "width": 128,
                      }
                    }
                  />
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#171717",
                        "elevation": 2,
                        "fontFamily": undefined,
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "letterSpacing": 0,
                        "lineHeight": 20,
                        "marginLeft": 8,
                        "marginTop": 4,
                        "position": "absolute",
                        "shadowColor": "#000000",
                        "shadowOffset": {
                          "height": 1,
                          "width": 0,
                        },
                        "shadowOpacity": 0.2,
                        "shadowRadius": 1.41,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    study
                  </Text>
                  <View
                    dataSet={{}}
                    style={
                      {
                        "marginLeft": -12,
                        "marginTop": 80,
                      }
                    }
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#dc2626",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      3
                      h left!
                    </Text>
                  </View>
                </View>
              </View>
              <View
                dataSet={{}}
                style={{}}
              />
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
                testID="checkInBox"
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "paddingBottom": 20,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 20,
                      "width": 150,
                    }
                  }
                >
                  <Image
                    accessibilityLabel="Alternate Text"
                    alt="Alternate Text"
                    dataSet={{}}
                    onError={[Function]}
                    source={
                      {
                        "uri": "https://picsum.photos/200",
                      }
                    }
                    style={
                      {
                        "borderRadius": 12,
                        "height": 128,
                        "maxWidth": "100%",
                        "position": "absolute",
                        "width": 128,
                      }
                    }
                  />
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#171717",
                        "elevation": 2,
                        "fontFamily": undefined,
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "letterSpacing": 0,
                        "lineHeight": 20,
                        "marginLeft": 8,
                        "marginTop": 4,
                        "position": "absolute",
                        "shadowColor": "#000000",
                        "shadowOffset": {
                          "height": 1,
                          "width": 0,
                        },
                        "shadowOpacity": 0.2,
                        "shadowRadius": 1.41,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    sleep
                  </Text>
                  <View
                    dataSet={{}}
                    style={
                      {
                        "marginLeft": -12,
                        "marginTop": 80,
                      }
                    }
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#dc2626",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      3
                      h left!
                    </Text>
                  </View>
                </View>
              </View>
              <View
                dataSet={{}}
                style={{}}
              />
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
                testID="checkInBox"
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "paddingBottom": 20,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 20,
                      "width": 150,
                    }
                  }
                >
                  <Image
                    accessibilityLabel="Alternate Text"
                    alt="Alternate Text"
                    dataSet={{}}
                    onError={[Function]}
                    source={
                      {
                        "uri": "https://picsum.photos/200",
                      }
                    }
                    style={
                      {
                        "borderRadius": 12,
                        "height": 128,
                        "maxWidth": "100%",
                        "position": "absolute",
                        "width": 128,
                      }
                    }
                  />
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#171717",
                        "elevation": 2,
                        "fontFamily": undefined,
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "letterSpacing": 0,
                        "lineHeight": 20,
                        "marginLeft": 8,
                        "marginTop": 4,
                        "position": "absolute",
                        "shadowColor": "#000000",
                        "shadowOffset": {
                          "height": 1,
                          "width": 0,
                        },
                        "shadowOpacity": 0.2,
                        "shadowRadius": 1.41,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    fitness
                  </Text>
                  <View
                    dataSet={{}}
                    style={
                      {
                        "marginLeft": -12,
                        "marginTop": 80,
                      }
                    }
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#dc2626",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      3
                      h left!
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;
